TribalWars/src/ApplicationCore/Interfaces/Repository/ITribeRepository.cs

```diff
@@ -0,0 +1,16 @@
+﻿using ApplicationCore.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ApplicationCore.Interfaces.Repository
+{
+    public interface ITribeRepository
+    {
+        int CreateTribe(Tribe tribe);
+        Task AddPlayerToTribe(TribePlayer player);
+        Task<Tribe> SelectedTribe(string tribeName);
```

Comment:
This interface implementation returns int, void and Tribe, but interface suggests that last two methods are returning Tasks, usually suggesting an asynchronous call. 
Modify your interface to return 'void' and 'Tribe' or use full async/await flow in controller+service+repository classes.

Reply:
Done